### ISSUE TITLE
RuntimeError: cuda runtime error (30) on Ubuntu18,CUDA9.1,cudnn7.0.5 when torch.cuda.is_available() returns True

### ISSUE THREAD

**Issue by henryliu00** [pytorch/pytorch#9536]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description
"torch.cuda.is_available()" returns True, but when I call ".to(device)" it cause following error:

THCudaCheck FAIL file=/opt/conda/conda-bld/pytorch_1524590031827/work/aten/src/THC/THCTensorRandom.cu line=25 error=30 : unknown error
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/henryliu/anaconda3/lib/python3.6/site-packages/torch/cuda/__init__.py", line 161, in _lazy_init
    torch._C._cuda_init()
RuntimeError: cuda runtime error (30) : unknown error at /opt/conda/conda-bld/pytorch_1524590031827/work/aten/src/THC/THCTensorRandom.cu:25

## Code example
```
>>> import torch
>>> torch.cuda.is_available()
True
>>> device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
>>> device
device(type='cuda', index=0)
>>> x = torch.tensor([[1]])
>>> x
tensor([[1]])
>>> x.to(device)
THCudaCheck FAIL file=/opt/conda/conda-bld/pytorch_1524590031827/work/aten/src/THC/THCTensorRandom.cu line=25 error=30 : unknown error
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/henryliu/anaconda3/lib/python3.6/site-packages/torch/cuda/__init__.py", line 161, in _lazy_init
    torch._C._cuda_init()
RuntimeError: cuda runtime error (30) : unknown error at /opt/conda/conda-bld/pytorch_1524590031827/work/aten/src/THC/THCTensorRandom.cu:25
```

## System Info

Collecting environment information...
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: 9.1.85

OS: Ubuntu 18.04 LTS
GCC version: (Ubuntu 4.8.5-4ubuntu8) 4.8.5
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.1.85
GPU models and configuration: GPU 0: GeForce GTX 1060 with Max-Q Design
Nvidia driver version: 390.48
cuDNN version: Probably one of the following:
/usr/local/cuda-9.1/lib64/libcudnn.so.7.0.5
/usr/local/cuda-9.1/lib64/libcudnn_static.a

Versions of relevant libraries:
[pip] numpy (1.14.3)
[pip] numpydoc (0.8.0)
[pip] torch (0.4.0)
[pip] torchvision (0.2.1)
[conda] cuda91                    1.0                  h4c16780_0    pytorch
[conda] pytorch                   0.4.0           py36_cuda9.1.85_cudnn7.1.2_1  [cuda91]  pytorch
[conda] torchvision               0.2.1                    py36_1    pytorch

**Comment by henryliu00** [comment]
Thank you so much! It worked! I didn't realize it could be a root issue(my bad). Now I would just run everything related with sudo.

**Comment by Kelym** [comment]
> maybe "sudo python" can solve it

Hey may I know why in this case we need to 'sudo python'? I am working on a server where I don't have root.

**Comment by Ivvvy-895** [comment]
> @Kelym one of the issues (of error 30) is that the /dev/nvidia* files are not yet correctly initialized. When you use CUDA, it tries to initialize them properly. However, for it to touch the /dev folder, it needs super-user access.
> Maybe you can ask your sysadmin to run a small example GPU program under root, so that the driver-level initialization finished correctly.

my sysadmin run a script using cuda with root, but it still cannot work without root. May I ask what gpu program may solve this problem? or is there any other solutions?

**Comment by Ricardozzf** [comment]
@Ivvvy-895 
According to [link](https://discuss.pytorch.org/t/runtimeerror-cuda-error-unknown-error/22634/5), you should  reinstall nvidia driver completely, and if you have some wrong with reinstall driver, you can refer to [it](https://ubuntuforums.org/showthread.php?t=2337077&page=2).

**Comment by charles-bu** [comment]
I meet the same issue of that. 

when I checked the 
torch.cuda.is_available()
True

however, if I set the 
a =torch.Tensor([1]).cuda()

  File "C:\Users\puqia\Anaconda3\lib\site-packages\torch\cuda\__init__.py", line 162, in _lazy_init
    torch._C._cuda_init()

RuntimeError: cuda runtime error (30) : unknown error at ..\aten\src\THC\THCGeneral.cpp:87

my systems is WIN10, 
pytorch is 1.0

cuda is 10.0

**Comment by Miladiouss** [comment]
sudo rmmod nvidia_uvm
sudo rmmod nvidia
sudo modprobe nvidia
sudo modprobe nvidia_uvm

**Comment by charles-bu** [comment]
hi, 

thank you so much for your support. 

however, I may still not be able to use your command, as my system is WIN10.

meanwhile, I spent last 2 days to get the problem solved. 

The problem may be caused by my updating the Nvidia driver last month. Since it could bring more trouble to roll back to the older version, I kept updating the Nvidia these days. Once the March 6 version has been updated, the pytorch looks a little bit recovered. It can work after my starting the Geforce application. the GPU tensor could be setup correctly. 

I thought it might be some problem laid in the GPU tensor initialization. I double checked that the WIN10 env variable. I noticed that almost all those CUDA_SDK_PATH were not found any more in my system. therefore, I filled in those CUDA_SDK_PATH env variable in the register table. 

after that, it works nicely till now.

thanks again for your help.

Best regard,


Charles
